### PR TITLE
[DNM] Wip libcephfs richacl

### PR DESCRIPTION
**Pull request by ukernel**

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Please check this out when you're back @jcsp, since you were checking the last one. ;)

**Comment by jcsp** [comment]
The cmake part isn't working for me, I have richacl-devel installed but my CMakeCache.txt ends up with "WITH_RICHACL:BOOL=OFF" and there's no output about it to stdout while running cmake.  I'm just assuming that it's supposed to be on by default, or should I have to explicitly enable it?

File: src/common/config_opts.h [hunk 1]
```diff
@@ -356,6 +356,8 @@ OPTION(client_debug_inject_tick_delay, OPT_INT, 0) // delay the client tick for
 OPTION(client_max_inline_size, OPT_U64, 4096)
 OPTION(client_inject_release_failure, OPT_BOOL, false)  // synthetic client bug for testing
 OPTION(client_inject_fixed_oldest_tid, OPT_BOOL, false)  // synthetic client bug for testing
+OPTION(client_acl_type, OPT_STR, "")
```
Comment (jcsp) [inline src/common/config_opts.h]: Let's add something in /doc/ about what the allowable values are here

**Comment by jcsp** [comment]
Looks like we require a very recent (unreleased?) version of richacl to get your recent changes from upstream.  Can you work out what version we should depend on in RPM/DEB manifests?

**Comment by jcsp** [comment]
> The cmake part isn't working for me, I have richacl-devel installed but my CMakeCache.txt ends up with "WITH_RICHACL:BOOL=OFF" and there's no output about it to stdout while running cmake. I'm just assuming that it's supposed to be on by default, or should I have to explicitly enable it?

Oh right, I have to do "cmake -DWITH_RICHACL=ON"

**Comment by jcsp** [comment]
Needs rebase

**Comment by jcsp** [comment]
I've sent an email to andreas of https://github.com/andreas-gruenbacher/richacl to ask about release plans.

**Comment by jcsp** [comment]
OK, the tagged v1.11 version upstream should have the calls that this patch requires.

The Fedora package (for Fedora 24) has been updated to build 1.11 (http://pkgs.fedoraproject.org/cgit/rpms/richacl.git/commit/?h=f24&id=561296128fc52e16c0824fa773fc7b6c888b926f)

Next step is to get this package available in enough of our build/test environment, then we can merge this.

@ukernel it might still be a while, but would you mind rebasing on latest master so that this will be ready to try out when we have the environmental dependencies in place?

**Comment by ukernel** [comment]
rebased

**Comment by liewegas** [comment]
@batrick @ukernel time to resurrect this?  are the libs available upstream now?

**Comment by batrick** [comment]
If the work required to resurrect this is low then I'm okay with it. @ukernel?

**Pull request closed without merging**